### PR TITLE
Fct daily scheduled trips

### PR DESCRIPTION
**Pull request by lauriemerrell**
# Description

Progress towards #1761. Creates a `fct_daily_scheduled_trips` table. Also creates `int_gtfs_schedule__daily_scheduled_service_index` (and, in the process, `int_gtfs_schedule__long_calendar.sql`) to support further service and trip summary work. For example the current (v1) `fact_daily_service` table contains information about first stop, last stop on that service. I think that would be part of a `fct_daily_scheduled_service` table in the mart based on `fct_daily_scheduled_trips`. We can add further summary information in follow-up PRs, but this creates the skeleton. 

I think we will need to clean up/iterate the interplay between mart and intermediate tables here, but I think that can come with time and isn't prohibitive.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?

dbt run & dbt test on the two new tables succeeds.

### CODE REVIEW

File: warehouse/models/mart/gtfs/fct_daily_scheduled_trips.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        t1.feed_key,
+        t1.service_id,
+        t2.key AS trip_key,
+        t3.key AS route_key,
+        t2.warning_duplicate_primary_key AS warning_duplicate_trip_primary_key
+    FROM int_gtfs_schedule__daily_scheduled_service_index AS t1
+    LEFT JOIN dim_trips AS t2
+        ON t1.feed_key = t2.feed_key
+        AND t1.service_id = t2.service_id
+    LEFT JOIN dim_routes AS t3
```
Comment (atvaccaro) [inline warehouse/models/mart/gtfs/fct_daily_scheduled_trips.sql:31]: this is just a general style request, could aliases be made meaningful or just not use them? it's hard to read the `SELECT` and know where each column is actually coming from. I realize I haven't asked this until now.

File: warehouse/models/intermediate/gtfs/int_gtfs_schedule__long_calendar.sql [hunk 2]
```diff
@@ -0,0 +1,30 @@
+-- TODO: make an intermediate calendar and use that instead of the dimension
+WITH dim_calendar AS (
+    SELECT *
+    FROM {{ ref('dim_calendar') }}
+),
+
+int_gtfs_schedule__long_calendar AS (
+    -- Note that you can unnest values easily in SQL, but getting the column names
+    -- is weirdly hard. To work around this, we just UNION ALL.
```
Comment (atvaccaro) [inline warehouse/models/intermediate/gtfs/int_gtfs_schedule__long_calendar.sql:9]: just curious had you tried https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#unpivot_operator?

File: warehouse/models/mart/gtfs/_mart_gtfs.yml [hunk 3]
```diff
@@ -1117,3 +1117,34 @@ models:
...
+      it means that that `trip_key` was scheduled to occur on that `service_date`.
+      Specifically, it means that the associated `service_id` was active and had service
+      scheduled. Dates where a `service_id` was active but not scheduled (for example,
+      weekend dates within a weekday service's effective dates) are not listed in this table.
+    columns:
+      - name: key
+        tests:
+          - unique:
+              where: "not warning_duplicate_trip_primary_key"
+          - not_null
```
Comment (atvaccaro) [inline warehouse/models/mart/gtfs/_mart_gtfs.yml:1133]: nit: I usually like putting `not_null` first as it's a stronger test than unique; unique excludes nulls

File: warehouse/models/intermediate/gtfs/int_gtfs_schedule__daily_scheduled_service_index.sql [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+boolean_calendar_dates AS (
+    SELECT
+        date,
+        feed_key,
+        service_id,
+        CASE
+            WHEN exception_type = 1 THEN TRUE
+            WHEN exception_type = 2 THEN FALSE
+        END AS service_indicator
```
Comment (atvaccaro) [inline warehouse/models/intermediate/gtfs/int_gtfs_schedule__daily_scheduled_service_index.sql:28]: are there exception types other than 1 and 2? I think `exception_type = 1 AS service_indicator` would be equivalent.

also nit: if it's a boolean could we prefix with a verb? for example `has_service` or something

**Review by atvaccaro — APPROVED** [review]
no blockers

**Comment by atvaccaro** [comment]
Would it be possible to get a qualitative assessment of trip counts from @edasmalchi or someone else? Just curious how we match up against some current analyses.

File: warehouse/models/mart/gtfs/fct_daily_scheduled_trips.sql (hunk 1 above)
Comment (lauriemerrell) [reply warehouse/models/mart/gtfs/fct_daily_scheduled_trips.sql]: https://github.com/cal-itp/data-infra/pull/1936/commits/446c4290d2f02dea6d9f1e2eddfcde6212c1ae76

File: warehouse/models/intermediate/gtfs/int_gtfs_schedule__long_calendar.sql (hunk 2 above)
Comment (lauriemerrell) [reply warehouse/models/intermediate/gtfs/int_gtfs_schedule__long_calendar.sql:9]: This is copy-pasted from our existing long calendar table, didn't want to reinvent the wheel: https://github.com/cal-itp/data-infra/blob/main/warehouse/models/gtfs_views/gtfs_schedule_stg_calendar_long.sql. Can add a TODO comment

File: warehouse/models/mart/gtfs/_mart_gtfs.yml (hunk 3 above)
Comment (lauriemerrell) [reply warehouse/models/mart/gtfs/_mart_gtfs.yml:1133]: I think this is in this order basically everywhere in our entire codebase, I am disinclined to change it here unless you really want me to

File: warehouse/models/intermediate/gtfs/int_gtfs_schedule__daily_scheduled_service_index.sql (hunk 4 above)
Comment (lauriemerrell) [reply warehouse/models/intermediate/gtfs/int_gtfs_schedule__daily_scheduled_service_index.sql:28]: No other exception types are valid. https://gtfs.org/schedule/reference/#calendar_datestxt. But it's possible there could be an invalid value, not sure we'd want to just lump that with `exception_type = 2`. Current `not_null` test would catch this.

File: warehouse/models/intermediate/gtfs/int_gtfs_schedule__long_calendar.sql (hunk 2 above)
Comment (lauriemerrell) [reply warehouse/models/intermediate/gtfs/int_gtfs_schedule__long_calendar.sql:10]: https://github.com/cal-itp/data-infra/pull/1936/commits/acab7027b6ec5f1bc6ac5fac1fa2230b3c174557

File: warehouse/models/intermediate/gtfs/int_gtfs_schedule__daily_scheduled_service_index.sql (hunk 4 above)
Comment (lauriemerrell) [reply warehouse/models/intermediate/gtfs/int_gtfs_schedule__daily_scheduled_service_index.sql]: https://github.com/cal-itp/data-infra/pull/1936/commits/20d7da2a4f6d0f88665e7edb417d7e096117a8fd

**Comment by lauriemerrell** [comment]
> Would it be possible to get a qualitative assessment of trip counts from @edasmalchi or someone else? Just curious how we match up against some current analyses.

How strongly do you feel about doing this pre-merge?

**Comment by lauriemerrell** [comment]
Confirmed with @atvaccaro he's ok to merge

**Comment by atvaccaro** [comment]
> > Would it be possible to get a qualitative assessment of trip counts from @edasmalchi or someone else? Just curious how we match up against some current analyses.
> 
> How strongly do you feel about doing this pre-merge?

Nope go for it, it might just be useful at some point to diff against people's assumptions in v1 (`gtfs_schedule_fact_daily_trips`?)

**Pull request merged**